src/lib/code-sample-editor.ts

```diff
@@ -83,8 +84,26 @@ export class CodeSampleEditor extends LitElement {
   @property({attribute: false})
   files?: SampleFile[];
 
-  // TODO: make a public property/method to select a file
-  @property({attribute: false})
+  /**
+   * The name of the project file that is currently being displayed.
+   */
+  @property()
+  filename?: string;
```

Comment:
This property looks like it's settable externally and internally, which makes me a little wary (consider this element used in a lit-html template with the property being bound - it could get out of sync with the dirty-checking). Document that it's set when changing tabs, and doesn't reflect to the attribute.

Reply:
Done.